sql/core/src/test/resources/sql-tests/inputs/postgreSQL/select_having.sql

```diff
@@ -49,6 +49,7 @@ SELECT 1 AS one FROM test_having HAVING a > 1;
 SELECT 1 AS one FROM test_having HAVING 1 > 2;
 SELECT 1 AS one FROM test_having HAVING 1 < 2;
 
+-- [SPARK-33008] Spark SQL throws an exception
```

Comment:
I think this test case is wrong -- the next line says that it's to prove that we aren't scanning the table, but the condition `1 < 2` actually _does_ force us to scan the table. I think this test case should have used `HAVING 1 > 2`. Could you correct the test case and remove the comment?

Reply:
IIUC, the `HAVING` condition here is performed on the globally grouped result set. I think this query is to prove the row count of the global group does not affect the output of an always-true `HAVING` clause.

Reply:
I found that postgres returns correct results without errors for both `1 < 2` and `1 > 2`. While Spark SQL always scans the table and throws exceptions cause its optimizer does not take this case into account.

Reply:
Interesting! Would you mind filing a ticket for that?